### PR TITLE
travis packaging

### PR DESCRIPTION
**Pull request by travipross**
Updates some packaging things. `setup.cfg` is now used for static info while `setup.py` is still maintained for dynamic content (as distinguished [here](https://packaging.python.org/en/latest/tutorials/packaging-projects/#configuring-metadata)).

I've switched from `python setup.py sdist` to a more generic `python -m build` as recommended [here](https://packaging.python.org/en/latest/tutorials/packaging-projects/#generating-distribution-archives), which reads the `pyproject.toml` file to determine which build backend to use (still `setuptools` in our case).

Also of note, I added a license file as it's highly recommended to use. Given that you've made this public and shared it, I think MIT license is most appropriate. [Here](https://choosealicense.com/licenses/mit/#) is some more info if you'd like to consider this further and possibly choose a different one.

Lastly, I added `isort` to the Makefile to help keep code tidy and consistent. No CI enforcement, but I could add that too if you'd like. **EDIT**: I added it to CI.

Some helpful links:
- https://towardsdatascience.com/packaging-in-python-tools-and-formats-743ead5f39ee
- https://choosealicense.com/
- https://packaging.python.org/en/latest/tutorials/packaging-projects/#packaging-python-projects
- https://setuptools.pypa.io/en/latest/userguide/quickstart.html

### CODE REVIEW

File: poglink/__init__.py [hunk 1]
```diff
@@ -1 +1,7 @@
+from importlib.metadata import version, PackageNotFoundError
 from . import cogs
+
+try:
+    __version__ = version("poglink")
+except PackageNotFoundError:
+    pass
```
Comment (travipross) [inline poglink/__init__.py:7]: Allows you to do:
```python
import poglink 

print(poglink.__version__)
```

See [here](https://github.com/pypa/setuptools_scm#retrieving-package-version-at-runtime) for reference

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+license_file = LICENSE
+project_urls =
+    Bug Tracker = https://github.com/fm-17/poglink/issues
+    DockerHub = https://hub.docker.com/r/fm17/poglink
+classifiers = 
+    Development Status :: 4 - Beta
+    License :: OSI Approved :: MIT License
+    Operating System :: POSIX :: Linux
+    Programming Language :: Python :: 3.7
+    Framework :: AsyncIO
```
Comment (travipross) [inline setup.cfg:19]: A hasty selection of classifiers I found from [here](https://pypi.org/classifiers/), feel free to suggest additions,removals,modifications etc. Just thought it'd be "proper" to include.

File: setup.py [hunk 3]
```diff
@@ -13,26 +13,8 @@
...
     use_scm_version = True
 
-with open("README.md") as f:
-    long_description = f.read()
-
 setup(
-    name="poglink",
     version=version,
     use_scm_version=use_scm_version,
     setup_requires=["setuptools_scm"],
```
Comment (travipross) [inline setup.py:19]: Pretty barebones now, just needs to be here for legacy support and for this dynamic stuff.

File: tests/integration/test_cogs_bans.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
-import pytest
 import aiohttp
-from poglink.cogs.bans import Bans
+import pytest
+
 from poglink.bot import ConfigurableBot
+from poglink.cogs.bans import Bans
```
Comment (travipross) [inline tests/integration/test_cogs_bans.py:5]: This import sorting comes from running `make isort`

File: scripts/python-publish.sh [hunk 5]
```diff
@@ -1,9 +1,6 @@
 #!/bin/bash
 set -e 
 
-# Build python source dist
-python setup.py sdist 
-
```
Comment (travipross) [inline scripts/python-publish.sh:6]: This script was meant to publish, not to build. Separating that logic to avoid any confusion.

**Review by FM-17 — APPROVED** [review]

**Pull request merged**